app/lib/month.rb

```diff
@@ -66,9 +66,9 @@ def self.current
   def self.parse_month(month)
     return month if month.is_a?(Month)
 
-    month_params = month.split('-').first(2)
-    Month.new(*month_params)
-  rescue ArgumentError, NoMethodError
+    match = /^(\d{4})-(\d{2})(?:-\d{2})?$/.match(month)
+    Month.new(*match.captures) if match
```

Comment:
Returns `nil` if the value does not match the regex, which is in line with the behavior for other "wrong input" values